src/target/ext/host_io.rs

```diff
@@ -345,7 +361,7 @@ pub trait HostIoReadlink: HostIo {
         &mut self,
         filename: &[u8],
         output: HostIoOutput<'a>,
-    ) -> HostIoResult<u32, Self>;
+    ) -> HostIoResult<HostIoToken<'a>, Self>;
```

Comment:
Could you update the docs for this method with a brief blurb explaining what this `HostIoToken` is all about?

e.g:

```
/// The data read _must_ be sent by calling `HostIoOutput::write`, which will consume the `output` object and return a 
/// `HostIoToken`. This token ensures that the implementer of this method calls `HostIoOutput::write`.
```